plugins/store/src/store.rs

```diff
@@ -361,38 +427,45 @@ impl<R: Runtime> std::fmt::Debug for StoreInner<R> {
 }
 
 pub struct Store<R: Runtime> {
-    defaults: Option<HashMap<String, JsonValue>>,
-    serialize: SerializeFn,
-    deserialize: DeserializeFn,
     auto_save: Option<Duration>,
     auto_save_debounce_sender: Arc<Mutex<Option<UnboundedSender<AutoSaveMessage>>>>,
     store: Arc<Mutex<StoreInner<R>>>,
 }
 
-impl<R: Runtime> Resource for Store<R> {}
+impl<R: Runtime> Resource for Store<R> {
+    fn close(self: Arc<Self>) {
+        let store = self.store.lock().unwrap();
+        let state = store.app.state::<StoreState>();
+        let mut stores = state.stores.lock().unwrap();
+        stores.remove(&store.path);
+    }
+}
 
 impl<R: Runtime> Store<R> {
-    pub fn with_store<T>(
-        &self,
-        f: impl FnOnce(&mut StoreInner<R>) -> crate::Result<T>,
-    ) -> crate::Result<T> {
+    /// Do something with the inner store,
+    /// useful for batching some work if you need higher performance
+    pub fn with_store<T>(&self, f: impl FnOnce(&mut StoreInner<R>) -> T) -> T {
```

Comment:
sounds like we need a better API, exposing `StoreInner` is not a good thing IMO

Reply:
I think exposing it here is fine, the reason for the outer `Store` is really just because of auto save and shared references, the inner store is the actual APIs and the outer store is just a convenient wrapper

Reply:
I really don't think it should be exposed, inner structs should always be internal, surface API should be refactor and changed to hide it.

Reply:
Even if `InnerStore` and `Store` share the same APIs, never expose it in a user API. It will only cause confusion and won't allow us to modify this type freely.

Reply:
I don't want this either to be honest, but people might want higher performance operations or if they want to batch some modifications atomically

What about we remove this function for now and see if people actually need it

Reply:
sure, if we want to do patch operations, we may want to rethink the API with a different one.